bga_database/base_models.py

```diff
@@ -1,8 +1,31 @@
 from django.db import models
+from django.db.models import Q
 
 
 class SluggedModel(models.Model):
     slug = models.SlugField(max_length=255, unique=True, null=True)
 
     class Meta:
         abstract = True
+
+
+class AliasModel(models.Model):
+    name = models.CharField(max_length=255)
+    preferred = models.BooleanField(default=False)
+
+    class Meta:
+        abstract = True
+
+    def clean(self):
+        try:
+            preferred_aliases = type(self).objects.filter(
```

Comment:
`filter()` won't raise an exception unless you pass invalid filter parameters. I'd suggest doing this outside of a `try/except` block, then checking for the conditions using `if` statements.